minio/minio
# WIP: add support for SSE-S3 bulk ETag decryption

## Description
This commit adds experimental support for
bulk ETag decryption for SSE-S3 encrypted
objects.

Currently, this feature has to be enabled
using `MINIO_DEBUG_KMS_KES_BULK_API=on`.

If KES supports a bulk decryption API, then
MinIO will check whether its policy grants
access to this API. If so, MinIO will use
a bulk API call instead of sending encrypted
ETags serially to KES.

Note that MinIO will not use the KES bulk API
if its client certificate is an admin identity.

## Motivation and Context
KMS, performance, SSE-S3

## How to test this PR?
KES `v.0.19.0`. Setup MinIO+KES as explained here: https://github.com/minio/kes/wiki/MinIO-Object-Storage
Also set `MINIO_DEBUG_KMS_KES_BULK_API=on` and allow the `/v1/key/bulk/decrypt/*` API in the KES policy.

Then, MinIO should be able to use the bulk decryption API for object listing.
**This is an experimental implementation**

## Types of changes

## Checklist:
